components/ReadingChallenge/ReadingChallenge.js

```diff
@@ -0,0 +1,192 @@
+import styled from "styled-components";
+import { useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
+import ChallengeBookList from "../ChallengeBookList/ChallengeBookList";
+
+export default function ReadingChallenge({
+  books,
+  booksInfo,
+  handleToggleAlreadyRead,
+}) {
+  const [modalState, setModalState] = useState({
+    isOpen: false,
+    isCancelled: false,
+  });
```

Comment:
The `modalState` initial object includes `isOpen` and `isCancelled`, but `isSaved` is used in the JSX without being part of the state.  🤔

_Please ensure all modal-related states are managed consistently._